pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -633,6 +635,13 @@ func (cb *ClusterBuilder) buildLocalityLbEndpoints(proxyNetworkView map[network.
 	return localityLbEndpoints
 }
 
+func getEndpointHealthStatus(instance *model.ServiceInstance) v3.HealthStatus {
+	if instance.Endpoint.LbWeight == 0 {
```

Comment:
how can you tell if weight is `0` because user explicitly set it to 0 or because they did not set it at all?

I think this will mean all users must explicitly set `weight: 1` to work.

Reply:
We can have a flag for this feature based on this flag we'll know if WLE 0 weight means no traffic or default weight 1.
But I also think introduce the adminState in WLE is a better solution, it aligns with the Envoy LBEndpoint health status.